WooCommerce/Classes/POS/ViewModels/TotalsViewModel.swift

```diff
@@ -73,7 +76,9 @@ final class TotalsViewModel: ObservableObject {
 
     init(orderService: POSOrderServiceProtocol,
          cardPresentPaymentService: CardPresentPaymentFacade,
-         currencyFormatter: CurrencyFormatter) {
+         currencyFormatter: CurrencyFormatter,
+         paymentState: PaymentState = .acceptingCard,
+         isSyncingOrder: Bool = false) {
```

Comment:
By setting these two properties and its initial values on initialization we're now setting some default initial state, when before that it was stateless. Note how the only dependencies being injected are abstractions like the payments Façade, utils like the currency formatter, or the POS interfaces.

Do we want it to always default to `.acceptingCard` and not `syncing`? Would it be preferable to leave the implementation details to each service? Eg: Payment state being set and updated from the `CardPresentPaymentService` rather than injected in.

Reply:
Looks like these have [initial values on trunk](https://github.com/woocommerce/woocommerce-ios/blob/trunk/WooCommerce/Classes/POS/ViewModels/TotalsViewModel.swift#L27)?

Reply:
> Looks like these have [initial values on trunk](https://github.com/woocommerce/woocommerce-ios/blob/trunk/WooCommerce/Classes/POS/ViewModels/TotalsViewModel.swift#L27)?

They are, but it's also internal state to the class. A concern here would be: 

Imagine we implement a new XYZ that needs the `TotalsViewModel` to be in `.acceptingCard` state. Since the property is exposed, and this happens to be its default value then we just ignore setting the property specifically. Later on we update the implementation details of the class to be `.notAcceptingCard` instead, which will cause XYZ to fail because it relied on default state that is no longer the default case, and would trigger no errors (unless catch by unit tests!) because the compiler would find no issues with it.

To avoid these, we can just not expose the state to be modified from outside the class, and leave the responsibility of it to the `TotalsViewModel` itself, opening some getters so it can be checked (but not modified) for other objects that need to check it.

Reply:
That's a a good point.

Addressed in commit bb3d973460da54ed4d807a56d91978a735e88840.